clang/include/clang/Sema/Sema.h

```diff
@@ -10204,15 +10205,14 @@ class Sema final : public SemaBase {
   /// both @c a1 and @c a2. If @p SuppressUserConversions, then don't
   /// allow user-defined conversions via constructors or conversion
   /// operators.
-  void
-  AddMethodCandidate(CXXMethodDecl *Method, DeclAccessPair FoundDecl,
-                     CXXRecordDecl *ActingContext, QualType ObjectType,
-                     Expr::Classification ObjectClassification,
-                     ArrayRef<Expr *> Args, OverloadCandidateSet &CandidateSet,
-                     bool SuppressUserConversions = false,
-                     bool PartialOverloading = false,
-                     ConversionSequenceList EarlyConversions = std::nullopt,
-                     OverloadCandidateParamOrder PO = {});
+  void AddMethodCandidate(
+      CXXMethodDecl *Method, DeclAccessPair FoundDecl,
+      CXXRecordDecl *ActingContext, QualType ObjectType,
+      Expr::Classification ObjectClassification, ArrayRef<Expr *> Args,
+      OverloadCandidateSet &CandidateSet, bool SuppressUserConversions = false,
+      bool PartialOverloading = false,
+      ConversionSequenceList EarlyConversions = std::nullopt,
+      OverloadCandidateParamOrder PO = {}, VarDecl *LambdaName = nullptr);
```

Comment:
I dont think this should be a default argument, we need to make sure every use of this function passes along this lambda if it has it anyway, and I think this should be paired with 'Method' above.

Reply:
sorry, did not get around to this comment this weekend. to be certain, would we also not want `LambdaName` (now `LambdaDecl`) to be a default argument for  `addFunctionCandidates()` above?

Reply:
I would think not as well, but if while working on it there is a compelling reason not to, I could have my mind changed.